### DISCUSSION TITLE
YOLO8 version  n,s,m

### DISCUSSION THREAD

**Discussion by PetrBrejtr** [ultralytics/ultralytics#10078] [Q&A]
Dear all, I would like to ask you for help with choosing a version for training the model.
I would like to detect defect (see photo)
Can i use yolo8n? 

![detection](https://github.com/ultralytics/ultralytics/assets/117436209/a86fe854-5abb-436b-a5ab-665ef4e9f8d1)

thank you 
Petr

**Comment by glenn-jocher** [comment]
@PetrBrejtr hello Petr,

Yes, you can start with YOLOv8n for detecting defects as shown in your photo. YOLOv8n is designed to be lightweight and efficient, making it a suitable choice for getting started and for applications where speed is crucial. If you find YOLOv8n's accuracy meets your needs, that's great. If you require higher accuracy and your computational resources allow, you can later experiment with YOLOv8s or YOLOv8m for potentially better performance.

Here's a simple example to train a YOLOv8n model on your dataset:

```bash
yolo detect train data=your_dataset.yaml model=yolov8n.pt
```

Make sure to replace `your_dataset.yaml` with your dataset configuration file. Wishing you the best with your defect detection project! 🚀

Warm

**Reply by PetrBrejtr** [reply]
Thank you so much!!!
with best regards
Petr

**Reply by glenn-jocher** [reply]
@PetrBrejtr You're welcome, Petr! 😊 If you have any more questions or need further assistance, feel free to reach out. Happy coding!

**Comment by PetrBrejtr** [comment]
Dear all, 
I did training with YOLOv8n and i have the result (see photo).
it's good? 
![results](https://github.com/ultralytics/ultralytics/assets/117436209/c1934966-579e-4163-a764-62094a767a2b)

**Reply by pderrenger** [reply]
@PetrBrejtr hello Petr,

The results look quite promising! However, for a more definitive assessment, you might consider checking the precision, recall, and mAP metrics typically displayed during validation. These metrics will give you a better understanding of your model's performance across different object classes and IoU thresholds.

If anything seems off, tweaking hyperparameters or further training might be necessary. Keep up the great work! 🚀

**Reply by PetrBrejtr** [reply]
thank you very much for your help!!!

**Reply by pderrenger** [reply]
@PetrBrejtr you're welcome! If you have any more questions or need further assistance as you continue with your project, feel free to reach out. Happy detecting! 🚀
